### PR TITLE
Fix #2454 of new UI SDK

### LINKED ISSUE #2454: Consolidate banner and notification source of data
As part of `1.0` effort we should consolidate the source of data that we're using for the banner and the notification, as it's currently different 👀 https://github.com/mapbox/mapbox-navigation-android/pull/2044#issuecomment-529054075

cc @mapbox/navigation-android

### PR DESCRIPTION
**Pull request by JunDai**
## Description

Please include a brief summary of the change and which issue is fixed (e.g., fixes [#issue](link))

- [ ] I have added any issue links
- [ ] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [ ] I have added the appropriate milestone and project boards

### Goal

Please describe the PR goals. Just the stuff needed to implement the fix / feature and a simple rationale. It could contain many check points if needed

### Implementation

Please include all the relevant things implemented and also rationale, clarifications / disclaimers etc. related to the approach used. It could be as self code companion comments

## Screenshots or Gifs

Please include all the media files to give some context about what's being implemented or fixed. It's not mandatory to upload screenshots or gifs, but for most of the cases it becomes really handy to get into the scope of the feature / bug being fixed and also it's REALLY useful for UI related PRs screenshot gif

## Testing

Please describe the manual tests that you ran to verify your changes

- [ ] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [ ] I have tested via a test drive, or a simulation/mock location app
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have updated the `CHANGELOG` including this PR

### CODE REVIEW

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/InstructionView.java [hunk 1]
```diff
@@ -258,15 +258,17 @@ public void updateDistanceWith(RouteProgress routeProgress) {
   }
 
   public void updateBannerInstructionsWith(BannerInstructions instructions) {
-    if (instructions == null || instructions.primary() == null) {
-      return;
+    if (instructions != null) {
+      updateBannerInstructions(instructions.primary(),
```
Comment (Guardiola31337) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/InstructionView.java]: Is it 🆗 passing `null` here `instructions.primary()`? There was a `null` check before 🤔

Comment (JunDai) [reply libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/InstructionView.java]: good catch! Checked the code again, we should have that `null` check to keep safe.

**Comment by JunDai** [comment]
@Guardiola31337 - I think we have `MapboxTripNotificationTest` to cover the notification test. It should be sufficient IMO.

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/InstructionView.java (hunk 1 above)
Comment (Guardiola31337) [reply libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/InstructionView.java]: Could you add it to the legacy code as well 🙏? It's fine if you do it as part of this PR 👍

**Comment by Guardiola31337** [comment]
> I think we have `MapboxTripNotificationTest` to cover the notification test. It should be sufficient IMO.

Is `MapboxTripNotificationTest` covering all the cases tested in [`MapboxNavigationNotificationManeuverInfoTest`](https://github.com/mapbox/mapbox-navigation-android/blob/master/libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/internal/navigation/MapboxNavigationNotificationManeuverInfoTest.java)? If that's the case 👍 if not, I'd add it to the test suite, it's not harmful 😅

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/InstructionView.java (hunk 1 above)
Comment (JunDai) [reply libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/InstructionView.java]: see above, I have included the null pointer protection into the legacy code.

**Comment by JunDai** [comment]
> MapboxNavigationNotificationManeuverInfoTest

Yeah, I think we are good with `MapboxTripNotificationTest`. The original `MapboxNavigationNotificationManeuverInfoTest` verifies the `currentManeuverType` after updating the notification. I added the same verification into `MapboxTripNotificationTest`.

**Pull request merged**